File: C++/Demo/RailFenceDemo.cpp
/*
	File		RailFenceDemo.cpp
	Author		Cory Bohon <cory@corybohon.com>
	Date		April 18, 2012
	Version		1.0
	Purpose		To demonstrate the RailFence class
*/

#include "RailFence.h"
#include <iostream>
#include <string.h>

using namespace std; 
	
int main()
{
	//Create a new RailFence object
	RailFence *railFence = new RailFence();

	//Create the string containing the original text
	string original_text = "This is my name...";
	
	//Print out the original text
	cout << "ORIGINAL TEXT:" << original_text << endl;
	
	//Encrypt the original text, storing the return in a new variable
	string encrypted_text = (*railFence).encrypt(original_text);
	
	//Print out the encrypted text
	cout << "ENCRYPTED TEXT:" << encrypted_text << endl;
	
	//Decrypt the encrypted text, storing the return in a new variable 
	string decrypted_text = (*railFence).decrypt(encrypted_text);
	
	//Print out the decrypted text 
	cout << "DECRYPTED TEXT:" << encrypted_text << endl;
	
	return 0;
}

File: Java/RailFence Demo/RailFence Demo/src/RailFenceDemo.java
/*
 * @author		Cory Bohon <cory@corybohon.com>
 * 
 * Purpose		Demonstrate the RailFence cipher in action
 * 
 */

public class RailFenceDemo 
{

	public static void main(String[] args) 
	{
		//string to be encrypted
		String originalText = "This is my name...";
			
		//Print out the original string
		System.out.println("Original Text: \t\t" + originalText);
		
		//create a new RailFence object
		RailFence fence = new RailFence();
		
		//call the method encrypt in the RailFence class to encrypt the passed originalText 
		//store the return in the encryptedString variable
		String encryptedString = fence.encrypt(originalText);
		
		//Print out the encrypted text
		System.out.println("Encrypted Text: \t" + encryptedString);
		
		//call the method decrypt in the RailFence class to decrypt the passed encryptedText
		//store the return in the decryptedString variable 
		String decryptedString = fence.decrypt(encryptedString);
		
		//Print out the decrypted text
		System.out.println("Decrypted Text: \t" + decryptedString);
	}

}
